pkg/pillar/containerd/containerd.go

```diff
@@ -885,6 +901,10 @@ func (client *Client) UnpackClientImage(clientImage containerd.Image) error {
 
 // StartUserContainerdInstance execute user containerd instance in goroutine
 func StartUserContainerdInstance() error {
+	// In kubevirt env we do not start eve user containerd, we just use that comes with k3s.
+	if base.IsHVTypeKube() {
```

Comment:
I dislike this `if` statement here, but get why it may be necessary.  if we are running with kubevirt, then we are saying, "containerd already started, do not bother starting it. Is that it? Is there a way instead to flag, "containerd is here", something more generic?

Reply:
> for one thing we can't be sure that the other instance in kube is started yet at the point in bootup

This is a very valid point. If so, then the correct question wouldn't be, "is containerd already running", but rather "am I the right place to start it". That is what your logic does here, which makes sense.

The question I am asking is, is this the right place and the right way to phrase the logic. 

I think a cleaner implementation is to have some determination somewhere (probably a `func init(){..}`  clause), that checks logic such as "am I running kubevirt"), and then sets a whole bunch of vars. The rest of the funcs just return the set vars (e.g. line 89). This centralizes the setting of it, creates one place in the future to set things that differ, and keep the logic simple.